src/mod_private_mnesia.erl

```diff
@@ -81,13 +80,19 @@ get_data(LUser, LServer, NS, Default) ->
         [] -> Default
     end.
 
-remove_user(LUser, LServer) ->
+remove_user(_HostType, LUser, LServer) ->
     F = fun() ->
                 NSs = select_namespaces_t(LUser, LServer),
         [delete_record_t(LUser, LServer, NS) || NS <- NSs]
         end,
     mnesia:transaction(F).
 
+%% There is no optimized way to remove a domain.
+%% We expect, that domain removal process would call remove_user instead
+%% for each user.
```

Comment:
I don't think we can expect anything from non-existing code. I think that there will be a need to notify the caller whether `remove_user` will need to be called or not, not just return `ok`. But this is all guessing as the code is not there yet. Could you convert this to a TODO?

Reply:
I think JIRA is a better place for that :)